### PR TITLE
Add possibility to link to mastodon

### PR DESCRIPTION
**Pull request by Janfred**
This is a requirement for a conference I'm organizing since we have our own mastodon instance and some of our guests don't have Twitter or try to avoid it as good as they can.

Unfortunately FontAwesome does not include an icon for mastodon yet, so I took a font from the [Issue in the Font Awesome Repo](https://github.com/FortAwesome/Font-Awesome/issues/10897)

### CODE REVIEW

**Comment by Ana06** [comment]
@differentreality why did you labelled this as `need feedback`? Is this something not clear? 🤔 

I think it is a good idea.

File: app/assets/stylesheets/mastodon.css.scss [hunk 1]
```diff
@@ -0,0 +1,45 @@
+@font-face {
```
Comment (Ana06) [inline app/assets/stylesheets/mastodon.css.scss]: Is this files copied from somewhere? 🤔

Comment (Janfred) [reply app/assets/stylesheets/mastodon.css.scss]: I took the main parts from here:

https://github.com/FortAwesome/Font-Awesome/issues/10897#issuecomment-318878737

I just adjusted some paths.

**Comment by differentreality** [comment]
@Ana06 to get feedback from the rest on whether or not we want to add this (since there was no issue prior to this PR where we could have discussed).

**Comment by Janfred** [comment]
Here are two screenshots of the pages which are updated by this PR

screenshot_osem_mastodon_admin
screenshot_osem_mastodon_spashpage

**Comment by Ana06** [comment]
@differentreality 

> @Ana06 to get feedback from the rest on whether or not we want to add this (since there was no issue prior to this PR where we could have discussed).

ok! 👍 

I personally think that as we already supported Twitter, there shouldn't be reason to not include mastodon as it is like an "open source Twitter" and we love open source 😛 And if you don't fill the field it doesn't appear, so it also doesn't increase the complexity. 👍

**Comment by bear454** [comment]
While I _completely support_ adding mastodon integration; I'd prefer to wait until https://github.com/FortAwesome/Font-Awesome/issues/10897 is closed so we can simply reference `fa-mastodon` instead of adding another font download to our already-heavy asset payload.

**Comment by Janfred** [comment]
@bear454 I can totally understand why you want to wait.
I'll have a look upon the FontAwesome-Discussion and will edit my PR when a mastodon Icon is included in FontAwesome.

This PR was intended to be a reference if other users of osem also want to have mastodon. (And I don't want my personal improvements for our use case to stay private)

**Comment by Ana06** [comment]
@bear454 

> While I completely support adding mastodon integration; I'd prefer to wait until FortAwesome/Font-Awesome#10897 is closed so we can simply reference fa-mastodon instead of adding another font download to our already-heavy asset payload.

Do we know how long this will take? If we expect this to happen soon, then I'll also wait. If not I'll add it and create an issue to change it when it is possible. 😉

**Comment by Janfred** [comment]
I've rebased my branch to the new master. I don't know how long it will take to add a mastodon icon to font awesome, I haven't seen anything in the icon request that gives me the feeling it will be added soon.

**Comment by Ana06** [comment]
I really think that we should support mastodon and I would like to avoid that @Janfred have to rebase again. So I'll be up for merging this now then. We can update it afterwards when it is added to font awesome.

@bear454 what do you think?

**Comment by bear454** [comment]
Yeah, that sounds fine, but please open an issue about removing if/when  FA supports mastodon.

**Comment by Janfred** [comment]
@bear454 I am subscribed to the issue in the font awesome repo. I'll open an issue or a pull request at the moment they are releasing a version with a mastodon-icon.

**Comment by Janfred** [comment]
@Ana06 I've rebased to the current master. I'll open the issue.

**Pull request merged**